src/invariant/BucketListIsConsistentWithDatabaseTests.cpp

```diff
@@ -2,330 +2,291 @@
+struct SelectBucketListGenerator : public BucketListGenerator
 {
-    auto keys = deleteRandomLedgerEntries(n, allLive, gen);
-    for (auto const& key : keys)
+    uint32_t const mSelectLedger;
+    LedgerEntryType const mType;
+    std::shared_ptr<LedgerEntry> mSelected;
+
+    SelectBucketListGenerator(uint32_t selectLedger, LedgerEntryType type)
+        : mSelectLedger(selectLedger), mType(type)
     {
-        selected.erase(key);
     }
-    return keys;
-}
 
-template <typename ABW, typename T>
-void
-checkBucketListIsConsistentWithDatabase(
-    std::function<std::vector<EntryFrame::pointer>(uint32_t, LedgerEntryType,
-                                                   T&)>
-        genLive = generateValidEntryFrames,
-    std::function<std::vector<LedgerKey>(uint32_t, std::set<LedgerKey> const&,
-                                         std::default_random_engine&, T&)>
-        genDead = {})
-{
-    std::default_random_engine gen;
-    for (auto t : xdr::xdr_traits<LedgerEntryType>::enum_values())
+    virtual std::vector<LedgerEntry>
+    generateLiveEntries(AbstractLedgerState& ls)
     {
-        size_t nTests = 0;
-        while (nTests < 10)
+        if (mLedgerSeq == mSelectLedger)
         {
-            LedgerEntryType let = static_cast<LedgerEntryType>(t);
-
-            VirtualClock clock;
-            Application::pointer appGenerate =
-                createTestApplication(clock, getTestConfig(0));
-            Application::pointer appApply =
-                createTestApplication(clock, getTestConfig(1));
-
-            T selected;
-            uint32_t ledgerSeq = generateLedgers(
-                appGenerate, 2, 100, 5,
-                std::bind(genLive, _1, let, std::ref(selected)), 2,
-                std::bind(genDead, _1, _2, std::ref(gen), std::ref(selected)));
-
-            // NOTE: We only advance nTests if selected is not empty so that
-            // we make sure we actually performed a test.
-            if (!selected.empty())
+            std::set<LedgerKey> filteredKeys;
+            std::copy_if(
+                mLiveKeys.begin(), mLiveKeys.end(),
+                std::inserter(filteredKeys, filteredKeys.end()),
+                [this](LedgerKey const& key) { return key.type() == mType; });
+
+            if (!filteredKeys.empty())
             {
-                auto dist = std::uniform_int_distribution<size_t>(
-                    0, selected.size() - 1);
-                auto index = dist(gen);
-                auto iter = selected.cbegin();
-                std::advance(iter, index);
-                REQUIRE_THROWS_AS(
-                    applyBucketsAndCrankUntilDone<ABW>(appGenerate, appApply,
-                                                       ledgerSeq, *iter, let),
-                    InvariantDoesNotHold);
-                ++nTests;
+                std::uniform_int_distribution<uint32_t> dist(
+                    0, filteredKeys.size() - 1);
```

Comment:
need cast to `uint32_t` (from `size_t`):
conversion from 'size_t' to '_Ty', possible loss of data